dll/win32/shell32/CShellDispatch.cpp

```diff
@@ -357,25 +488,29 @@ HRESULT STDMETHODCALLTYPE CShellDispatch::ShowBrowserBar(BSTR clsid, VARIANT sho
 HRESULT STDMETHODCALLTYPE CShellDispatch::AddToRecent(VARIANT file, BSTR category)
 {
     TRACE("(%p, %s, %ls)\n", this, wine_dbgstr_variant(&file), category);
-    return E_NOTIMPL;
+
+    CComHeapPtr<ITEMIDLIST> idlist;
+    HRESULT hr = VariantToIdlist(&file, &idlist);
+    if (hr == S_OK)
+        SHAddToRecentDocs(SHARD_PIDL, (LPCITEMIDLIST)idlist);
+    else
+        hr = S_FALSE;
```

Comment:
Are you sure you want to force this `hr` return value to `S_FALSE` in case it isn't success? Maybe it's better to report the value as-is back to the caller? :thinking: 
```suggestion
    HRESULT hr = VariantToIdlist(&file, &idlist);
    if (hr == S_OK)
        SHAddToRecentDocs(SHARD_PIDL, (LPCITEMIDLIST)idlist);
```

Reply:
There are many places where the real error is masked as S_FALSE. Remember that this class mainly exists for scripting languages to use and returning real errors will stop the script unless the script is prepared to handle errors (try/catch in JScript).

I could not get Windows to fail this method by giving it bad input.